Source/Portal/Generators/AamsuBridge.cpp

```diff
@@ -55,32 +49,19 @@ void AamsuBridge::SpawnBridge()
 	GetWorld()->LineTraceSingleByChannel(HitResult, InitialBridgeTransform->GetComponentLocation(), CheckDestination,
 		ECollisionChannel::ECC_Visibility, FCollisionQueryParams);
 	
-	const float FullPartsLength = HitResult.Distance - FMath::Fmod(HitResult.Distance, BridgePartLength);	
-	
-	for (float Dist = 0; Dist < FullPartsLength; Dist += BridgePartLength)
-	{
-		UStaticMeshComponent* NewPart = DuplicateObject<UStaticMeshComponent>(BridgePart, this);
-		NewPart->RegisterComponent();
-	 	
-		NewPart->SetWorldTransform(TargetTransform);
+	const float BridgeLength = HitResult.Distance;
 
-		TargetTransform.AddToTranslation(InitialBridgeTransform->GetForwardVector() * BridgePartLength);
-
-		BridgeParts.Add(NewPart);
+	if (BridgeLength < 0.1f)
+	{
+		return;
 	}
-
-	// TODO Let's use scaling
+	FVector Scale = FVector::ZeroVector;
+	Scale.Y = BridgeLength/ 100.f;
```

Comment:
1) Move constant to UPROPERTY 
2)
```suggestion
	constexpr float DefaultLength = 100.f;
	Scale.Y = BridgeLength / DefaultLength;
```